apps/telemetry_event/lib/telemetry_event/helper/telemetry_event_helper.ex

```diff
@@ -10,20 +10,32 @@ defmodule TelemetryEvent.Helper.TelemetryEventHelper do
     ]
   end
 
-  def measurements(options) do
-    options
+  def tags_and_values(event_tags_and_values) do
+    event_tags_and_values
     |> Map.new(fn {k, v} -> {k, replace_nil(v)} end)
   end
 
   defp metrics() do
     Application.get_env(:telemetry_event, :metrics_options)
     |> Enum.map(fn metrics_option ->
-      Metrics.counter(Keyword.fetch!(metrics_option, :metric_name),
-        tags: Keyword.fetch!(metrics_option, :tags)
-      )
+      Keyword.fetch!(metrics_option, :metric_type)
+      |> String.upcase()
+      |> metrics_event(metrics_option)
     end)
   end
 
+  defp metrics_event(metric_type, metrics_option) when metric_type === "COUNTER" do
```

Comment:
By convention this is done like `defp metrics_event("COUNTER", metrics_option) do` to take advantage of Elixir's pattern matching.

In addition, we try to avoid hard-coded strings duplicated across apps. Look into either making them methods like we do with event names so that `telemetry_event` controls them entirely, or use atoms like `:counter`